rmw_connext_cpp/src/functions.cpp

```diff
@@ -230,7 +230,7 @@ rmw_create_publisher(
     }
   }
 
-  if (!get_datawriter_qos(participant, qos_profile, datawriter_qos)) {
+  if (!get_datawriter_qos(participant, *qos_profile, datawriter_qos)) {
```

Comment:
Should the Connext shared API be updated to use a pointer rather then a reference?